projects/packages/connection/src/class-tokens.php

```diff
@@ -487,16 +487,21 @@ public function update_blog_token( $token ) {
 	 *
 	 * @todo Refactor to properly load the XMLRPC client independently.
 	 *
-	 * @param Integer $user_id the user identifier.
-	 * @param bool    $can_overwrite_primary_user Allow for the primary user to be disconnected.
-	 * @return Boolean Whether the disconnection of the user was successful.
+	 * @param int  $user_id The user identifier.
+	 * @param bool $can_overwrite_primary_user Allow for the primary user to be disconnected, deprecated.
+	 *
+	 * @return bool Whether the disconnection of the user was successful.
 	 */
 	public function disconnect_user( $user_id, $can_overwrite_primary_user = false ) {
 		$tokens = $this->get_user_tokens();
 		if ( ! $tokens ) {
 			return false;
 		}
 
+		if ( false !== $can_overwrite_primary_user ) {
+			_deprecated_argument( __METHOD__, '$$next-version$$', 'Parameter $can_overwrite_primary_user is deprecated' );
+		}
+
 		if ( Jetpack_Options::get_option( 'master_user' ) === $user_id && ! $can_overwrite_primary_user ) {
```

Comment:
Do we still need this check here since we are deprecating `$can_overwrite_primary_user`?

Reply:
@fgiannar since we're deprecating the argument, and not removing it, we need to let consumers know that they aren't supposed to use it now.
So I think adding the `_deprecated_argument()` notice is a good safeguard, even though I doubt there are any consumers of the function that still use the argument.

Reply:
As discussed in Slack, we'll keep the argument for now, and use the deprecation notice to inform potential consumers of the upcoming change.